storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -1263,24 +1266,31 @@ private void setupStormCode(Map<String, Object> conf, String topoId, String tmpJ
         String codeKey = ConfigUtils.masterStormCodeKey(topoId);
         String confKey = ConfigUtils.masterStormConfKey(topoId);
         NimbusInfo hostPortInfo = nimbusHostPortInfo;
+
+        CuratorFramework zkClient = null;
```

Comment:
**`zkClient` must be guaranteed to be closed.** Please ensure `zkClient` is closed before end of method if zkClient is not null.